### PR TITLE
Sent Accept-Encoding: identity in the head/get requests.

### PR DESCRIPTION
**Pull request by qnikst**
It appeared that some s3 servers (Yandex storage in particular)
honour `Accept-Encoding: gzip` request header. In such a case
servers can't send `Content-Length` header as transfer size differ
from the body size.
The simplest solution for this problem is to force http-client
to send  `Accept-Encoding: identity` header in the HeadObject
request.

### CODE REVIEW

**Comment by qnikst** [comment]
@harshavardhana @donatello please check current approach, I've checked statObject and getObject for my files and it had worked well in my case.

File: src/Network/Minio/S3API.hs [hunk 1]
```diff
@@ -550,8 +551,8 @@ headObject bucket object reqHeaders = do
           riBucket = Just bucket,
           riObject = Just object,
           riHeaders = reqHeaders
+            <> [("Accept-Encoding", "identity")]
```
Comment (harshavardhana) [inline src/Network/Minio/S3API.hs:554]: Maybe add a comment here for future readers as reference would help, so that it's not removed by mistake.

Comment (qnikst) [reply src/Network/Minio/S3API.hs:560]: Fixed!

**Comment by kannappanr** [comment]
Just want to know if `minio-hs` is setting `Accept-Encoding` header to `gzip` by default?

**Comment by harshavardhana** [comment]
> Just want to know if `minio-hs` is setting `Accept-Encoding` header to `gzip` by default?

@kannappanr I think that it is default for some http clients, if not already set. In minio-go it's automatically disabled because we ask the client to not unzip the stream automatically.

**Pull request merged**